packages/eslint-plugin/tests/rules/prefer-function-type.test.ts

```diff
@@ -146,5 +146,78 @@ interface Foo<T> {
 type Foo<T> = (bar: T) => string;
       `,
     },
+    {
+      code: `
+interface Foo<T> {
+  (this: T): void;
+}
+      `,
+      errors: [
+        {
+          messageId: 'functionTypeOverCallableType',
+          type: AST_NODE_TYPES.TSCallSignatureDeclaration,
+        },
```

Comment:
I don't know why the rest of the tests don't do this, but could you please add `data` to assert the shape of the message is what we expect?

```suggestion
          type: AST_NODE_TYPES.TSCallSignatureDeclaration,
          data: {
            type: 'Foo',
          },
        },
```

Reply:
well the message contained the suggestion so it would have been redundant for the test to assert the fix and message suggestion, the correct behaviour is definitely to remove the suggestion from the message and assert the data.